p2p/src/net/default_backend/peer.rs

```diff
@@ -473,10 +481,13 @@ mod tests {
             TEST_PROTOCOL_VERSION.into(),
         );
 
-        let handle = tokio::spawn(async move {
-            peer.handshake(P2pTimestamp::from_int_seconds(123456)).await.unwrap();
-            peer
-        });
+        let handle = tokio::spawn(
+            async move {
+                peer.handshake(P2pTimestamp::from_int_seconds(123456)).await.unwrap();
+                peer
+            }
+            .instrument(tracing::Span::current()),
+        );
```

Comment:
The pattern of spawning a tokio task and instrumenting it with the current span is repeated many times. Would it make sense to factor it out?

Reply:
Yeah, I was also thinking about it. I'm not sure though, what's the best way to do it. I could just wrap `tokio::spawn`, factoring out the entire call to "instrument", or add a custom trait with a function like "forward_current_tracing_span", so that only the "tracing::Span::current()" part gets encapsulated.

Reply:
Yea I'm not sure. I would probably go either for wrapping spawn (something like `spawn_with_current_span`) or just leave it as it is, the repetition is not too bad after all. I don't think factoring out only the `Span` part would not be much different from repeating it since you still get the `spawn(future.forward_span())` patter repeated.

Reply:
So, I've added the function and named it `spawn_in_current_span`. There are two places though where I still have to instrument the futures explicitly, they are in `add_subsystem_with_config` where `JoinSet::spawn` is called instead of `tokio::spawn`. But it turned out that there is a more concise way of doing it - the extension trait that contains the `instrument` method also has `in_current_span`, which calls `self.instrument(Span::current())`.